src/compiler/rust/canister_methods/src/hash_file.rs

```diff
@@ -0,0 +1,131 @@
+            let limit = 60 * 1024 * 1024; // This limit will be determine by how much hashing an update method can do without running out of cycles. It runs out somewhere between 120 and 125
+            // This limit must be the same as on the node side or else the hashes will not match
+            let mut buffer = vec![0; limit];
+            let bytes_read = std::io::Read::read(&mut file, &mut buffer);
+
+            let previous_hash = get_partial_file_hash(&path);
+
+            match bytes_read {
+                Ok(bytes_read) => {
+                    if bytes_read != 0 {
+                        let new_hash = hash_chunk_with(&buffer, previous_hash.as_ref());
+                        set_partial_hash(&path, new_hash);
+                        let delay = core::time::Duration::new(0, 0);
+                        let closure = move || hash_file_by_parts(path, position + bytes_read as u64);
+                        ic_cdk_timers::set_timer(delay, closure);
+                    } else {
+                        // No more bytes to hash, set as final hash for this file
+                        match previous_hash {
+                            Some(hash) => {
+                                set_file_hash(&path, hash);
+                                ic_cdk::println!("Finish hashing");
+                                clear_partial_hash(&path);
+                                ic_cdk::println!("Partial hash cleared");
+                            }
+                            None => ic_cdk::println!("WARNING: No hash was found for {}", path),
+                        }
+                    }
+                }
+                Err(err) => panic!("Error reading file: {}", err),
+            }
+        }
+
+        pub fn get_partial_file_hash(path: &String) -> Option<Vec<u8>> {
+            PARTIAL_FILE_HASHES.with(|file_hashes| file_hashes.borrow().get(path).cloned())
+        }
+
+        fn set_partial_hash(path: &String, hash: Vec<u8>) {
+            PARTIAL_FILE_HASHES.with(|file_hashes| file_hashes.borrow_mut().insert(path.clone(), hash));
+        }
+
+        fn clear_partial_hash(path: &String) {
+            PARTIAL_FILE_HASHES.with(|file_hashes| file_hashes.borrow_mut().remove(path));
+        }
+
+        fn clear_file_hash(path: &String) {
+            let mut file_hashes = load_hashes().unwrap();
+            file_hashes.remove(path);
+            save_hashes(&file_hashes).unwrap();
+        }
+
+        fn set_file_hash(path: &String, hash: Vec<u8>) {
+            let mut file_hashes = load_hashes().unwrap();
+            file_hashes.insert(path.clone(), hash);
+            save_hashes(&file_hashes).unwrap();
+        }
+
+        fn hash_chunk_with(data: &[u8], previous_hash: Option<&Vec<u8>>) -> Vec<u8> {
+            let mut h = <sha2::Sha256 as sha2::Digest>::new();
```

Comment:
I think we can simplify the hashing, let mut h = Sha256::new()

h.update()

Reply:
I got you
`let mut h: sha2::Sha256 = sha2::Digest::new();`
And h.update() doesn't exist without importing

To my knowledge the reason we are Qualifying everywhere instead of Importing is to make it easier to manage imports since things are spread out among many almost unrelated files because of using the quotes. 

If this is not the case or we actually don't think that's important anymore and we would rather Import that would clean up a lot of stuff.